File: DarkOrbit 7.5.3/Net/netty/commands/AmmunitionTypeModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class AmmunitionTypeModule
    {
        public const short ID = 15629;

        public const short ROCKET = 0;

        public const short TORPEDO = 1;

        public const short WIZARD = 2;

        public const short PRANK = 3;

        public const short BATTERY = 4;

        public const short FIREWORK = 5;

        public const short FIREWORK_1 = 6;

        public const short FIREWORK_2 = 7;

        public const short FIREWORK_3 = 8;

        public const short MINE = 9;

        public const short MINE_EMP = 10;

        public const short MINE_SAB = 11;

        public const short MINE_DD = 12;

        public const short MINE_SL = 13;

        public const short SMARTBOMB = 14;

        public const short INSTANT_SHIELD = 15;

        public const short PLASMA = 16;

        public const short EMP = 17;

        public const short LASER_AMMO = 18;

        public const short ROCKET_AMMO = 19;

        public const short RSB = 20;

        public const short HELLSTORM = 21;

        public const short UBER_ROCKET = 22;

        public const short ECO_ROCKET = 23;

        public const short SAR01 = 24;

        public const short SAR02 = 25;

        public const short X1 = 26;

        public const short X2 = 27;

        public const short X3 = 28;

        public const short X4 = 29;

        public const short SAB = 30;

        public const short CBO = 31;

        public const short R310 = 32;

        public const short PLT2026 = 33;

        public const short PLT2021 = 34;

        public const short PLT3030 = 35;

        public const short BDR1211 = 36;

        public const short DECELERATION = 37;

        public const short CBR = 38;

        public const short HITAC_LASER = 39;

        public const short JOB100 = 40;

        public const short BDR1212 = 41;

        public short typeValue;

        public AmmunitionTypeModule(short typeValue)
        {
            this.typeValue = typeValue;
        }

        public byte[] write()
        {
            ByteArray param1 = new ByteArray(ID);
            param1.writeShort(this.typeValue);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/clientCommands/AmmunitionTypeModule.java
package simulator.netty.clientCommands;

import java.io.DataInputStream;
import java.io.IOException;

import simulator.netty.ClientCommand;

/**
 Description: This is the ammunition type module

 @author Ordepsousa
 @date 21/07/2014
 @file AmmunitionTypeModule.java
 @package game.objects.netty.ServerCommands */
public class AmmunitionTypeModule
        extends ClientCommand {

    public static int id = 15629;
    public short typeValue;
    public static final short ROCKET         = 0;
    public static final short TORPEDO        = 1;
    public static final short WIZARD         = 2;
    public static final short PRANK          = 3;
    public static final short BATTERY        = 4;
    public static final short FIREWORK       = 5;
    public static final short FIREWORK_1     = 6;
    public static final short FIREWORK_2     = 7;
    public static final short FIREWORK_3     = 8;
    public static final short MINE           = 9;
    public static final short MINE_EMP       = 10;
    public static final short MINE_SAB       = 11;
    public static final short MINE_DD        = 12;
    public static final short MINE_SL        = 13;
    public static final short SMARTBOMB      = 14;
    public static final short INSTANT_SHIELD = 15;
    public static final short PLASMA         = 16;
    public static final short EMP            = 17;
    public static final short LASER_AMMO     = 18;
    public static final short ROCKET_AMMO    = 19;
    public static final short RSB            = 20;
    public static final short HELLSTORM      = 21;
    public static final short UBER_ROCKET    = 22;
    public static final short ECO_ROCKET     = 23;
    public static final short SAR01          = 24;
    public static final short SAR02          = 25;
    public static final short X1             = 26;
    public static final short X2             = 27;
    public static final short X3             = 28;
    public static final short X4             = 29;
    public static final short SAB            = 30;
    public static final short CBO            = 31;
    public static final short R310           = 32;
    public static final short PLT2026        = 33;
    public static final short PLT2021        = 34;
    public static final short PLT3030        = 35;
    public static final short BDR1211        = 36;
    public static final short DECELERATION   = 37;
    public static final short CBR            = 38;
    public static final short HITAC_LASER    = 39;
    public static final short JOB100         = 40;
    public static final short BDR1212        = 41;

    /**
     Constructor
     */
    public AmmunitionTypeModule(DataInputStream in) {
        super(in, id);
    }

    /**
     Description: Reads command
     */
    public void readInternal() {
        try {
            typeValue = in.readShort();
        } catch (IOException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
    }
}